src/libraries/System.Private.CoreLib/src/System/String.cs

```diff
@@ -508,6 +508,7 @@ public static bool IsNullOrWhiteSpace([NotNullWhen(false)] string? value)
         public ref readonly char GetPinnableReference() => ref _firstChar;
 
         internal ref char GetRawStringData() => ref _firstChar;
+        internal ref ushort GetRawStringDataAsUshort() => ref Unsafe.As<char, ushort>(ref _firstChar);
```

Comment:
nit - Convention is:
```suggestion
        internal ref ushort GetRawStringDataAsUInt16() => ref Unsafe.As<char, ushort>(ref _firstChar);
```